web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/CdsProfileFetcher.java

```diff
@@ -39,7 +46,7 @@
 
 public class CdsProfileFetcher implements AppProfileFetcher {
 
-	private HttpAsyncClient httpClient;
+	private CloseableHttpAsyncClient httpClient;
```

Comment:
why shouldn't this variable be final? Do we ever need to do multiple initialization for this?

Reply:
The setter is public

Reply:
Sure in this case it cannot be fine but atleast @littleaj the setter can be set to package private (aka default access). This setter is only used in CdsProfileFetcherTests.java which resides in the same package. I know this is not part of your PR but it's good if we can address if possible in this PR.

Reply:
@dhaval24 I don't think we can reduce the visibility of a public method in a public class w/o going through the deprecation procedure.